Filter out tests directories in get_couchapps

gherceg opened the pull request:
## Product Description
<!-- Where applicable, describe user-facing effects and include screenshots. -->

## Technical Summary
<!--
    Provide a link to any tickets, design documents, and/or technical specifications
    associated with this change. Describe the rationale and design decisions.
-->
Followup to https://github.com/dimagi/commcare-hq/pull/37052 that actually filters out 'tests' directories to prevent looking in those directories for design docs.

## Feature Flag
<!-- If this is specific to a feature flag, which one? -->

## Safety Assurance

### Safety story
<!--
Describe how you became confident in this change, such as
local testing, why the change is inherently safe, and/or plans to limit the blast radius of a defect.

In particular consider how existing data may be impacted by this change.
-->
Slight modification to similar existing logic to filter out \_\_pycache\_\_. Started to add a test and decided it wasn't worth it.

### Automated test coverage

<!-- Identify the related test coverage and the tests it would catch -->

### QA Plan

<!--
- Describe QA plan that along with automated test coverages proves this PR is regression free
- Link to QA Ticket
-->
No

### Rollback instructions

<!--
If this PR follows standards of revertability, check the box below.
Otherwise replace it with detailed instructions or reasons a rollback is impossible.
-->

- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

gherceg committed:
Filter out tests directories in get_couchapps

snopoke reviewed (approved):
A more generic approach would be to check if the directory conforms to the couch spec: https://docs.couchdb.org/en/stable/ddocs/ddocs.html#creation-and-structure

gherceg committed:
Run makemigrations to request reindex

Due to the tests directory now being filtered out/changing the
couchviews.lock file, a migration was autodetected. The reindex should
be virtually a no-op.
